app/views/dradis/plugins/csv/upload/new.html.erb

```diff
@@ -12,70 +12,56 @@
-                <option disabled selected>Select a Field</option>
-                <option>Field form RTP</option>
-                <option>Another from RTP</option>
-                <option disabled>&#9472;</option>
-                <option>Create **COLUMN NAME GOES HERE** field</option>
-                <option>Do not import this column</option>
-              </select>
-              <label for="field_0" class="sr-only">Field in Dradis</label>
-            </div>
-          </td>
-          <td>
-            <div class="validation">
-              <%# if error %>
-                <i class="fa fa-times fa-fw ml-4" data-toggle="tooltip" data-original-title="Error message goes here"></i>
-              <%# else %>
-                <%# <i class="fa fa-check fa-fw ml-4" data-toggle="tooltip" data-original-title="Validation passed!"></i> %>
-              <%# end %>
-            </div>
-          </td>
-        <%# else %>
-          <%# <td>**COLUMN NAME GOES HERE**</td> %>
-        <%# end %>
-      </tr>
-    </tbody>
-  </table>
+  <%= form_with url: project_upload_index_path(current_project), method: :post do |f| %>
+    <table class="table table-striped mb-0"
+      data-behavior="dradis-datatable"
+      data-column-toggle="false"
+      data-default-columns="<%= @default_columns %>"
+      data-item-name="csv_item"
+      data-local-storage-key="project.ce.<%= dom_id(current_project) %>.csv_mapping_datatable"
+    >
+      <thead>
+        <tr>
+          <th class="no-sort">Unique Identifier <i class="fa fa-question-circle" data-toggle="tooltip" title="Ensure duplicate issues with this value are not created."></i></th>
+          <th>Column Header From File</th>
+          <th>Type</th>
+          <th>Field in Dradis</th>
+        </tr>
+      </thead>
+          <tbody>
+            <% @headers.each_with_index do |header, index| %>
+              <tr>
+                <td>
+                  <div class="form-check">
+                    <%= f.radio_button 'identifier', index, class: 'form-check-input' %>
+                    <%= f.label :identifier, class: 'form-check-label' do %>
+                      <span class="sr-only">Select</span>
+                    <% end %>
+                  </div>
+                </td>
+                <td><%= header %></td>
+                <td>
+                  <div class="form-group m-0">
+                    <%= f.select "mappings[#{index}][type]", ['Issue Field', 'Evidence Field', 'Node Label'], {}, class: 'form-control custom-select w-75', data: { behavior: 'type-select' } %>
+                  </div>
+                </td>
+                <td>
+                  <% if current_project.report_template_properties %>
+                    <div class="form-group m-0">
+                      <%= f.text_field "mappings[#{index}][field]", class: 'form-control' %>
+                    </div>
+                  <% else %>
+                    <span data-behavior="default-field-label"><%= header %></span>
```

Comment:
We still need a hidden_field with the header as the value so that users can submit the values to the controller.

Reply:
If we add a hidden_field, a user can edit that value manually to pass an unexpected parameter. Not a security issue, but I'd rather we not allow this. Can we just let the controller handle the logic if no field was passed?